model/parts/reserve.py

```diff
@@ -2,33 +2,36 @@
 Reserve metric and advanced balance calculation
 """
 
-from model.types import (
-    Fiat,
-    Pair
-)
-from model.utils.exchange_rates import get_stable_balance_in_usd
+
+from model.types import CryptoAsset
 
 
 def p_reserve_statistics(
-    params,
+    _params,
     _substep,
     _state_history,
     prev_state,
 ):
     """
     calculates reserve statistics
     """
-    reserve_balance_usd = sum([inventory * prev_state['market_price'].get(Pair(key, Fiat.USD))
-                           for key, inventory in prev_state['reserve_balance'].items()])
+    reserve_values_usd = prev_state['reserve_balance'].values_in_usd(
+        prev_state)
+    reserve_balance_usd = sum(list(reserve_values_usd.values()))
+
+    reserve_celo_usd = reserve_values_usd.get(CryptoAsset.CELO)
+
+    floating_supply_values_usd = prev_state['floating_supply'].values_in_usd(
+        prev_state)
+    floating_supply_balance_usd = sum(list(floating_supply_values_usd.values()))
 
-    stables_balance_usd = get_stable_balance_in_usd(prev_state['floating_supply'],
-                                                prev_state['market_price'],
-                                                params['mento_exchanges_config'])
+    reserve_ratio = reserve_celo_usd / floating_supply_balance_usd
```

Comment:
do you want to calculate the reserve_ratio as approximated on chain? That's what I thought we want to do and then the scaling by celo target weight would be missing

`reserve_ratio = reserve_celo_usd / target_weight_celo / floating_supply_balance_usd`

with target_weight_celo 0.5 I think

https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/stability/Reserve.sol#L548